evaluate/task/task-code-repair.go

```diff
@@ -0,0 +1,155 @@
+package task
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/evaluate/metrics"
+	"github.com/symflower/eval-dev-quality/language"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/model"
+	evaltask "github.com/symflower/eval-dev-quality/task"
+)
+
+// TaskCodeRepair holds the code repair task.
+type TaskCodeRepair struct {
+	// ResultPath holds the directory path where results should be written to.
+	ResultPath string
+
+	// Language holds the language for which the task should be evaluated.
+	Language language.Language
+	// Model holds the model which the task should be evaluated.
+	Model model.Model
+
+	// Logger holds the logger for this tasks.
+	Logger *log.Logger
+}
```

Comment:
@ruiAzevedo19 @bauersimon This is not relevant for this PR, but I'd still like to discuss this briefly.

Having this fields in the `TaskXXX` types is pretty unnecessary since we use these fields in a single `Run` method (plus 1-2 helper methods). That is, there is no state that needs to be held.

I was thinking about renaming the current `task.Context` to `task.ModelContext` since the model gets that context, and put the fields of `TaskXXX` into a `task.Context`. and `Task.Run` then gets a `task.Context` as argument.

Thoughts?

Reply:
sounds good! it is just a collection of arguments after all

Reply:
Alright, I'll create a PR based on this one.

Reply:
Yap, sounds good to me too